tools/te.py

```diff
@@ -41,7 +41,7 @@
 #ifdef TYPE_ERASED_DECLARATION
 
 // Type-erased interface for:
-struct ${struct_name}
+struct MIGRAPHX_EXPORT ${struct_name}
```

Comment:
I would prefer to have come from a variable we define at the top:

```python
import string, sys, re
trivial = [
    'std::size_t', 'instruction_ref', 'support_metric', 'const_module_ref',
    'bool', 'any_ptr'
]

export_macro = 'MIGRAPHX_EXPORT'
...
form = string.Template('''
#ifdef TYPE_ERASED_DECLARATION
// Type-erased interface for:
struct ${struct_name}
struct ${export_macro} ${struct_name}
...
''')
...
def generate_form(name, members):
    nonvirtual_members = []
    pure_virtual_members = []
    virtual_members = []
    comment_members = []
    default_members = []
    decl_members = []
    for member in members:
        m = convert_member(member, name)
        nonvirtual_members.append(nonvirtual_member.substitute(m))
        pure_virtual_members.append(pure_virtual_member.substitute(m))
        virtual_members.append(virtual_member.substitute(m))
        comment_members.append(comment_member.substitute(m))
        decl_members.append(decl_member.substitute(m))
        if 'default' in m:
            default_members.append(default_member.substitute(m))
    return form.substitute(nonvirtual_members=''.join(nonvirtual_members),
                           pure_virtual_members=''.join(pure_virtual_members),
                           virtual_members=''.join(virtual_members),
                           default_members=''.join(default_members),
                           decl_members=''.join(decl_members),
                           comment_members='\n'.join(comment_members),
                           struct_name=name,
                           export_macro=export_macro)
```

Reply:
Done